### PR TITLE
UI: Remove use of React.FC in components

### PR DESCRIPTION
**Pull request by ShaunEvening**
## What I did

In React 18, the React team removed the children typing from `React.FC` and is now best practice not to use it over all.

With that breaking change, a bunch of our component's types make them unusable with children as the types are incorrect or too narrow. This aims to resolve that.

- [x] Remove most use of React.FC from our component types	
- [x] Make types for new Icon map better

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/ui/components/src/components/form/input/input.tsx [hunk 1]
```diff
@@ -207,7 +207,7 @@ type TextareaProps = Omit<
   valid?: ValidationStates;
   height?: number;
 } & React.RefAttributes<HTMLTextAreaElement>;
-export const Textarea: FC<TextareaProps> = Object.assign(
+export const Textarea = Object.assign(
   styled(
     forwardRef<any, TextareaProps>(function Textarea({ size, valid, align, ...props }, ref) {
```
Comment (kylegach) [inline code/ui/components/src/components/form/input/input.tsx:212]: Is `children` included somewhere in `TextareaProps`? I didn't see it.

Comment (Integrayshaun) [reply code/ui/components/src/components/form/input/input.tsx:212]: No but it shouldn't need it anyway for an input Element, right?

**Review by kylegach — APPROVED** [review]
Not sure about the comprehensiveness here (i.e. if there are other instances to address), but the refactors I see here look great!

File: code/ui/components/src/components/form/input/input.tsx (hunk 1 above)
Comment (kylegach) [reply code/ui/components/src/components/form/input/input.tsx:212]: Oh, the name threw me. I thought it was rendering a `<textarea>`.

Comment (kylegach) [reply code/ui/components/src/components/form/input/input.tsx:212]: Indeed, it is:

```ts
type TextareaPropsRaw = React.TextareaHTMLAttributes<HTMLTextAreaElement>;
```

**Comment by Integrayshaun** [comment]
@kylegach in the components package there's only the tabs component remaining because it does some children manipulation with pretty hairy types.

We should refactor that separately or deprecate it for an easier API

File: code/ui/components/src/components/form/input/input.tsx (hunk 1 above)
Comment (JReinhold) [reply code/ui/components/src/components/form/input/input.tsx:212]: @kylegach I don't think passing `children` to this `Textarea` is valid anyway, so it seems like this fixes a type bug, removing that possibility?

**Review by JReinhold — APPROVED** [review]

**Pull request merged**